File: Irrgarten_Ruby/Irrgarten.rb
#encoding:utf-8

require_relative 'Weapon'
require_relative 'Shield'
require_relative 'Dice'

module Directions
	LEFT =:left
	RIGHT=:right
	UP=:up
	DOWN=:down
end

module Orientation
	VERTICAL=:vertical
	HORIZONTAL=:horizontal
end

module GameCharacter
	PLAYER =:player
	MONSTER =:monster
end

module Irrgarten
	
	class Irrgarten
		
		def self.prueba_weapon
			w = Weapon.new(2.0, 3)
			puts "\n"
			puts w.to_s	
			4.times do
				puts "Potencia de disparo #{w.attack}"
				puts w.to_s
			end
		end
		
		def self.prueba_shield
			s = Shield.new(2.0,3)
			puts "\n"
			puts s.to_s
			4.times do
				puts "Daño protegido #{s.protect}"
				puts s.to_s
			end
		end
		
		def self.prueba_dice
			puts "\n"
			puts "Posición: #{Dice.random_pos(10)}"
			puts "Empieza jugador: #{Dice.who_starts(3)}"
			puts "Inteligencia: #{Dice.random_intelligence}"
			puts "Fuerza: #{Dice.random_strength}"
			puts "¿Resucita? #{Dice.resurrect_player}"
			puts "Armas conseguidas: #{Dice.weapons_reward}"
			puts "Escudos conseguidos: #{Dice.shields_reward}"
			puts "Salud conseguida: #{Dice.health_reward}"
			puts "Poder arma: #{Dice.weapon_power}"
			puts "Poder escudo: #{Dice.shield_power}"
			usos = Dice.uses_left
			puts "Usos restantes: #{usos}"
			puts "Intensidad: #{Dice.intensity(3.0)}"
			puts "¿Se descarta? #{Dice.discard_element(usos)}"
		end
		
	end

	#Irrgarten.prueba_weapon

end

Irrgarten::Irrgarten.prueba_weapon
Irrgarten::Irrgarten.prueba_shield
Irrgarten::Irrgarten.prueba_dice

File: Irrgarten/src/irrgarten/Irrgarten.java
package irrgarten;

enum Directions {
    LEFT, RIGHT, UP, DOWN
}

enum Orientation {
    VERTICAL, HORIZONTAL
}

enum GameCharacter {
    PLAYER, MONSTER
}
public class Irrgarten {

    static void pruebaWeapon (float daño, int usos){
        Weapon arma1 = new Weapon(daño,usos);
        System.out.println("Dañó: " + arma1.attack());
        
        System.out.println(arma1.toString());
    }
    
    static void pruebaShield(float durabilidad, int usos) {
        Shield escudo1 = new Shield(durabilidad,usos);
        System.out.println("Protegio: " + escudo1.protect());
        
        System.out.println(escudo1.toString());
    }
    
    static void pruebaDice(){
        Dice dado = new Dice();
        System.out.println("Posición: " + dado.randomPos(10));
        System.out.println("Empieza jugador " + dado.whoStarts(3));
        System.out.println("Inteligencia " + dado.randomIntelligence());
        System.out.println("Fuerza " + dado.randomStrength());
        System.out.println("¿Resucita? " + dado.resurrectPlayer());
        System.out.println("Armas conseguidas: " + dado.weaponsReward());
        System.out.println("Escudos conseguidos: " + dado.shieldsReward());
        System.out.println("Poder arma: " + dado.weaponPower());
        System.out.println("Poder escudo: " + dado.shieldPower());
        int usos = dado.usesLeft();
        System.out.println("Usos restantes: " + usos);
        System.out.println("Intensidad: " + dado.intensity(3.0f));
        System.out.println("¿Se descarta? " + dado.discardElement(usos));
    }
            
    public static void main(String[] args) {
       System.out.println("Probando arma:");
       pruebaWeapon(10.0f,1);
       pruebaWeapon(10.0f,0);
       
       System.out.println("\n Probando escudo:");
       pruebaShield(10.0f,1);
       pruebaShield(10.0f,0);
       
       System.out.println("\n Probando dado:");
       pruebaDice();

    }
    
}
